staging/src/k8s.io/apimachinery/pkg/util/wait/wait_test.go

```diff
@@ -939,7 +939,7 @@ func TestPollImmediateUntilWithContext(t *testing.T) {
 				defer cancel()
 				return ctx, cancel
 			},
-			errExpected:      ErrWaitTimeout,
```

Comment:
Did you search through existing uses to see if any need to be changed?

Reply:
No broken uses in our tree.  However, there are a few broken users on github.  I think for that reason I'll create new methods and mark the old ones deprecated.  Opinion on naming scheme to create the new correct ones?

Scenario that would be impacted:

1. User calls PollImmediateUntilWithContext() with a condition that returns no error
2. User checks the error value and *requires* it to equal ErrWaitTimeout (vs handling err != nil as a catch all), otherwise it does "something different"
3. The behavior of that "something different" is such that any other error besides ErrWaitTimeout means the condition failed with an error
4. The condition failing with the error vs being cancelled has a different meaning.

If the user simply handles the error generically this has no impact.  If the user treats any error except ErrWaitTimeout as "stop early", this has no impact.  Only if the user assumes `err != nil && err != ErrWaitTimeout`  means "the condition errored" will this have an impact.  That should be rare, but it's definitely possible.

Search https://cs.github.com/?scopeName=All+repos&scope=&q=%22.PollImmediateWithContext%22+ErrWaitTimeout had 11 hits, 6 of which are the same code:

* https://cs.github.com/alebedev87/node-observability-operator/blob/6c749f38b63cd2a9b63c158c27314d7d63812a9a/pkg/operator/controller/machineconfig/mcp.go?q=%22.PollImmediateWithContext%22+ErrWaitTimeout#L257 does not check for non ErrWaitTimeout errors, so a context cancellation would be treated as a nil error.

Reply:
Considering `*WithContextCancellable`.